internal/controller/pdbwatcher_controller.go

```diff
@@ -3,311 +3,117 @@ package controllers
-	logger := log.FromContext(ctx)
-
-	// Fetch the PDBWatcher instance
-	pdbWatcher := &myappsv1.PDBWatcher{}
-	err := r.Get(ctx, req.NamespacedName, pdbWatcher)
-	if err != nil {
-		//should we use a finalizer to scale back down on deletion?
-		if errors.IsNotFound(err) {
-			return ctrl.Result{}, nil // PDBWatcher not found, could be deleted, nothing to do
-		}
-		return ctrl.Result{}, err // Error fetching PDBWatcher
-	}
-
-	//only do this on create? kill off by sharing name?
-	if err := r.conflicts(ctx, pdbWatcher); err != nil {
-		return ctrl.Result{}, err
-	}
+// ToDo: Implement this?
+func (P PDBWatcher) DeepCopyObject() runtime.Object {
+	return nil
+}
 
-	// Fetch the PDB
-	pdb := &policyv1.PodDisruptionBudget{}
-	err = r.Get(ctx, types.NamespacedName{Name: pdbWatcher.Spec.PDBName, Namespace: pdbWatcher.Namespace}, pdb)
-	if err != nil {
-		//better error on notfound
-		return ctrl.Result{}, err // Error fetching PDB
-	}
+// PDBWatcherSpec defines the desired state of PDBWatcher
+type PDBWatcherSpec struct {
+	Name string `json:"name"`
+}
 
-	deploymentName := pdbWatcher.Spec.TargetName
-	if deploymentName == "" {
-		deploymentName, err := r.discoverDeployment(ctx, pdb) //move this out of thie controller to a controller that watches pdbs
-		if err != nil {
-			//better error on notfound
-			return ctrl.Result{}, err // Error fetching PDB
-		}
-		pdbWatcher.Spec.TargetName = deploymentName
-		pdbWatcher.Spec.TargetKind = deploymentKind
-		err = r.Update(ctx, pdbWatcher)
-		if err != nil {
-			logger.Error(err, "Failed to update PDBWatcher deployment")
-			return ctrl.Result{}, err
-		}
-		return ctrl.Result{}, nil
+// PDBWatcherStatus defines the observed state of PDBWatcher
+type PDBWatcherStatus struct {
+	Status string `json:"status"`
+}
 
-	}
+// PDBWatcherReconciler reconciles a PodDisruptionBudget object.
+type PDBWatcherReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
 
-	// Fetch the Deployment
-	target, err := GetSurger(pdbWatcher.Spec.TargetKind)
-	if err != nil {
-		return ctrl.Result{}, err
-	}
-	err = r.Get(ctx, types.NamespacedName{Name: pdbWatcher.Spec.TargetName, Namespace: pdbWatcher.Namespace}, target.Obj())
+// Reconcile reads the state of the cluster for a PDB and creates/deletes PDBWatchers accordingly.
+func (r *PDBWatcherReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	// Fetch the PodDisruptionBudget object based on the reconcile request
+	var pdb policyv1.PodDisruptionBudget
+	err := r.Get(ctx, req.NamespacedName, &pdb)
 	if err != nil {
-		if errors.IsNotFound(err) {
-			//blank out deployment and try again?
-			pdbWatcher.Spec.TargetName = ""
-			err = r.Update(ctx, pdbWatcher)
+		// If the PDB is deleted, we should delete the corresponding PDBWatcher.
+		// First, check if the PDBWatcher exists
+		var pdbWatcher PDBWatcher
+		err := r.Get(ctx, req.NamespacedName, &pdbWatcher)
```

Comment:
so we are assumign same name here. Document that if so.
in future changge we could get more aggressive about enforcing pdbwatcher always has same name as pdb.